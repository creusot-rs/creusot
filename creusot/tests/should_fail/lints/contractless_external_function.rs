extern crate creusot_contracts;
use creusot_contracts::*;

#[deny(creusot::contractless_external_function)]
#[requires(true)]
pub fn f() {
    // We will probably never specify `transmute`, so it is a good target to showcase this lint
    let _x: i32 = unsafe { std::mem::transmute(1i32) };
}
