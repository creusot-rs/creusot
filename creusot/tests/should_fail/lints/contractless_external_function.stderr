error: calling external function `transmute` with no contract will yield an impossible precondition
 --> contractless_external_function.rs:8:28
  |
8 |     let _x: i32 = unsafe { std::mem::transmute(1i32) };
  |                            ^^^^^^^^^^^^^^^^^^^^^^^^^ function called here
  |
note: the lint level is defined here
 --> contractless_external_function.rs:4:8
  |
4 | #[deny(creusot::contractless_external_function)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

